src/gwproto/data_classes/hardware_layout.py

```diff
@@ -197,9 +198,65 @@ def load_nodes(
                 errors.append(LoadError("ShNode", node_dict, e))
         return nodes
 
+    @classmethod
+    def make_channel(cls, dc_dict: dict, nodes: dict[str, ShNode]) -> DataChannel:
+        about_node_name = dc_dict.get("AboutNodeName")
+        captured_by_node_name = dc_dict.get("CapturedByNodeName")
+        about_node = nodes.get(about_node_name)
+        captured_by_node = nodes.get(captured_by_node_name)
+        return DataChannel(
+            about_node=about_node, captured_by_node=captured_by_node, **dc_dict
+        )
+
+    @classmethod
+    def load_data_channels(
+        cls,
+        layout: dict[Any, Any],
+        nodes: dict[str, ShNode],
+        *,
+        raise_errors: bool = True,
+        errors: Optional[list[LoadError]] = None,
+        included_channel_names: Optional[set[str]] = None,
+    ) -> dict[str, DataChannel]:
+        dcs = {}
+        if errors is None:
+            errors = []
+        for dc_dict in layout.get("DataChannels", []):
+            try:
+                dc_name = dc_dict["Name"]
+                if included_channel_names is None or dc_name in included_channel_names:
+                    dcs[dc_name] = cls.make_channel(dc_dict, nodes)
+            except Exception as e:  # noqa: PERF203
+                if raise_errors:
+                    raise
+                errors.append(LoadError("DataChannel", dc_dict, e))
+        return dcs
+
+    @classmethod
+    def resolve_dcs(
```

Comment:
I don't think we need this  if `make_channel` throws an exception when a node is None (as above).

Reply:
yep.